ci/build_container/build_recipes/yaml-cpp.sh

```diff
@@ -4,11 +4,26 @@ set -e
 
 VERSION=0.6.2
 
-wget -O yaml-cpp-"$VERSION".tar.gz https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-"$VERSION".tar.gz
+curl https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-"$VERSION".tar.gz -sLo yaml-cpp-"$VERSION".tar.gz
 tar xf yaml-cpp-"$VERSION".tar.gz
 cd yaml-cpp-yaml-cpp-"$VERSION"
-cmake -DCMAKE_INSTALL_PREFIX:PATH="$THIRDPARTY_BUILD" \
-  -DCMAKE_CXX_FLAGS:STRING="${CXXFLAGS} ${CPPFLAGS}" \
-  -DCMAKE_C_FLAGS:STRING="${CFLAGS} ${CPPFLAGS}" \
-  -DCMAKE_BUILD_TYPE=RelWithDebInfo .
-make VERBOSE=1 install
+
+mkdir build
+cd build
+
+if [[ "${OS}" == "Windows_NT" ]]; then
+  cmake -G"Ninja" -DCMAKE_INSTALL_PREFIX:PATH="$THIRDPARTY_BUILD" \
+    -DCMAKE_CXX_FLAGS:STRING="${CXXFLAGS} ${CPPFLAGS}" \
+    -DCMAKE_C_FLAGS:STRING="${CFLAGS} ${CPPFLAGS}" \
+    -DYAML_CPP_BUILD_TESTS=off \
+    -DCMAKE_BUILD_TYPE=Debug ..
+  ninja
+  ninja install
```

Comment:
Maybe we could switch to ninja on non-Window? I would prefer greater consistency for maintainability here.

Reply:
From our perspective, we are explicitly trying not to change anything about how non-Windows builds happen in this PR. Perhaps a future PR?

Also, switching to ninja on non-Windows introduces a new build dependency, which is maybe not good?

Reply:
I have a more fundamental question while we're thinking about this. How are we going to do CI, gate submissions and do build regression handling for Windows? I know a while back we chatted (@mattklein123 ) and had some idea that Windows would be best-effort, with the Microsoft folks owning the regression fixes. Is this where we are still at? I don't think CircleCI has Windows support.

Reply:
We spoke with @mattklein123 about this a couple of weeks ago. It seems like Windows support would still be best-effort at this time. We've used AppVeyor for Windows CI in the past, and that's what we were thinking of using. We could help get this set up so that the automation is in place, but not gate on CI passing (for now).